### PR TITLE
Fix optional param in PHPdoc for WC_Admin_Marketplace_Promotions

### PR DESCRIPTION
**Pull request by tomalec**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Fix optional param in PHPdoc for WC_Admin_Marketplace_Promotions
Optional params where marked with a convention not recognized by phpdoc and threw a warning in code-reference generator.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Generate [code reference](https://github.com/woocommerce/code-reference) for the latest release: `./deploy.sh -v --build-only -s 9.1.4`
2. Checkout this branch and build woocommerce.zip: 
```sh
git checkout dev/fix-phpdoc-code-reference
cd plugins/woocommerce
pnpm run build:zip
```
4. unzip file to the code-reference directory
5. Generate again from local changes ``./deploy.sh -v --build-only -s 9.1.4 --no-download``

##### Before
image

##### After 
image

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
Fix optional param in PHPdoc for `WC_Admin_Marketplace_Promotions` to generate code-reference w/o warnings
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/includes/admin/class-wc-admin-marketplace-promotions.php [hunk 1]
```diff
@@ -219,8 +219,8 @@ function ( $marketplace_pages ) use ( $promotion ) {
 	/**
 	 * From the array of promotions, select those of a given format.
 	 *
-	 * @param ? array  $promotions  Array of data about promotions of all formats.
-	 * @param ? string $format      Format we want to filter for.
+	 * @param array  $promotions  Array of data about promotions of all formats.
+	 * @param string $format      Format we want to filter for.
```
Comment (eason9487) [inline plugins/woocommerce/includes/admin/class-wc-admin-marketplace-promotions.php]: The parsing error doesn't seem to be caused by the question mark but by the space character in `? array`.

There are still a few `@param ?` in the codebase such as the following and they won't cause parsing errors.

https://github.com/woocommerce/woocommerce/blob/99e6a540d74a469209f216a9c1c862e3a82a77f0/plugins/woocommerce/includes/admin/class-wc-admin-marketplace-promotions.php#L287-L291

image

**Review by eason9487 — COMMENTED** [review]
I think the parsing error is caused by the space character between the question mark and the parameter datatype because the error was gone after I changed these lines to `?array` and `?string` in https://github.com/woocommerce/woocommerce/pull/50732/files#diff-99f2137654a5709ffd276fea599e758bbc5ce33eed69b0402939095e8bba4980L222-L223.

Therefore, for all changes, I would like to check if the fix should remove `?` or remove the redundant space.

File: plugins/woocommerce/includes/admin/class-wc-admin-marketplace-promotions.php (hunk 1 above)
Comment (tomalec) [reply plugins/woocommerce/includes/admin/class-wc-admin-marketplace-promotions.php]: Thanks, I didn't realize that. Brought back the `?` and removed the spaces in https://github.com/woocommerce/woocommerce/pull/50732/commits/b0a9b018431fb97e724e39ab37e4278a29fd4968

Comment (tomalec) [reply plugins/woocommerce/includes/admin/class-wc-admin-marketplace-promotions.php]: Thanks @eason9487 , I didn't realize that. Brought back the `?` and removed the spaces in https://github.com/woocommerce/woocommerce/pull/50732/commits/b0a9b018431fb97e724e39ab37e4278a29fd4968

**Review by eason9487 — APPROVED** [review]
Thanks for the fix. I can generate code reference without parsing errors. LGTM.

**Pull request merged**